lib/practice_exercises.rb

```diff
@@ -1,13 +1,35 @@
+# sources:
+# https://stackoverflow.com/questions/17689155/how-to-ignore-nonword-charcters-in-ruby
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) where n depends on size of .gsub?
+# Space Complexity: O(n) where n depends on size of .gsub?
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  string = string.gsub(/\W/, "").downcase # modify string without non word characters and spaces
+  x = 0
+  i = (string.length - 1)
+  until x >= i
+    if string[x] != string[i]
+      return false
+    end
+    x += 1
+    i -= 1
+  end
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(nm) where n is strings size, m is smallest word size?
+# Space Complexity: 0(n) where n is smallest word size?
 def longest_prefix(strings)
```

Comment:
👍  Although I would have said O(m) for space complexity just to keep the variables consistent.